utility/environment_check.py

```diff
@@ -0,0 +1,128 @@
+    defaults.ENV_STATUS_PRE['cephfs'] = CEPHFILESYSTEM.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_PRE['cephbp'] = CEPHBLOCKPOOL.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_PRE['pv'] = PV.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_PRE['pvc'] = PVC.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_PRE['secret'] = SECRET.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_PRE['namespace'] = NS.get(all_namespaces=True)['items']
+
+
+@pytest.fixture(scope='class')
+def get_status_after_execution():
+    """
+
+    """
+    defaults.ENV_STATUS_POST['pod'] = POD.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_POST['sc'] = SC.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_POST['cephfs'] = CEPHFILESYSTEM.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_POST['cephbp'] = CEPHBLOCKPOOL.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_POST['pv'] = PV.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_POST['pvc'] = PVC.get(all_namespaces=True)['items']
+    defaults.ENV_STATUS_POST['secret'] = SECRET.get(
+        all_namespaces=True
+    )['items']
+    defaults.ENV_STATUS_POST['namespace'] = NS.get(
+        all_namespaces=True
+    )['items']
+    pod_diff = DeepDiff(
+        defaults.ENV_STATUS_PRE['pod'], defaults.ENV_STATUS_POST['pod']
+    )
+    sc_diff = DeepDiff(
+        defaults.ENV_STATUS_PRE['sc'], defaults.ENV_STATUS_POST['sc']
+    )
+    cephfs_diff = DeepDiff(
+        defaults.ENV_STATUS_PRE['cephfs'], defaults.ENV_STATUS_POST['cephfs']
+    )
+    cephbp_diff = DeepDiff(
+        defaults.ENV_STATUS_PRE['cephbp'], defaults.ENV_STATUS_POST['cephbp']
+    )
+    pv_diff = DeepDiff(
+        defaults.ENV_STATUS_PRE['pv'], defaults.ENV_STATUS_POST['pv']
+    )
+    pvc_diff = DeepDiff(
+        defaults.ENV_STATUS_PRE['pvc'], defaults.ENV_STATUS_POST['pvc']
+    )
+    secret_diff = DeepDiff(
+        defaults.ENV_STATUS_PRE['secret'], defaults.ENV_STATUS_POST['secret']
+    )
+    namespace_diff = DeepDiff(
+        defaults.ENV_STATUS_PRE['namespace'],
+        defaults.ENV_STATUS_POST['namespace']
+    )
+    diffs_list = [
+        {'pods': pod_diff}, {'sc': sc_diff}, {'cephfs': cephfs_diff},
+        {'cephbp': cephbp_diff}, {'pvs': pv_diff}, {'pvcs': pvc_diff},
+        {'secret': secret_diff}, {'ns': namespace_diff},
+    ]
+    leftover_detected = False
+    leftovers = {'Leftovers added': [], 'Leftovers removed': []}
+    for kind in diffs_list:
+        if ADDED_RESOURCE in kind[''.join(kind.keys())]:
+            leftovers['Leftovers added'].append({
+                ''.join(kind.keys()):
+                kind[''.join(kind.keys())][ADDED_RESOURCE][
+                    ''.join(kind[''.join(kind.keys())][ADDED_RESOURCE])
+                ]
+            })
+            leftover_detected = True
```

Comment:
if this one is True, there is no reason to check the removed resources. I suggest checking if there are removed resources only in case there are no added resources

Reply:
however, you still want to check if there is anything to append to leftovers['Leftovers removed'] even if the previous leftover detection result was true.
So maybe having to leftover_detected args, one for added and one for removed

Reply:
I'm not sure I follow:
regarding the first comment:
I can create a secret and not delete it (leftover added) and delete a pod from the env by mistake (leftover removed).
Regarding the second comment:
I will print everything in any case (for kind in diffs_list)

Reply:
@RazTamir See my edit to the first comment

Reply:
@RazTamir my mistake :) in case there are added resources and no removed ones, leftover_detected will remain True

Reply:
> if this one is True, there is no reason to check the removed resources. I suggest checking if there are removed resources only in case there are no added resources.
A scenario where I created a resource and forgot to delete it will result in added_resource.
If I also deleted some other resource so I want to print it as well.
I think I'm missing something
> Actually, if there are added leftovers and no removed leftovers, leftover_detected will be False.
Why?
I'm not re-setting levftovers_detected to False ever again - after the first catch of leftover, it will print it
Same, I think I miss something

Reply:
> if this one is True, there is no reason to check the removed resources. I suggest checking if there are removed resources only in case there are no added resources.
> Actually, if there are added leftovers and no removed leftovers, leftover_detected will be False.

I read it that `leftover_detected` variable covers both addition and removal of something. This could be noted eg. at line:

```
leftover_detected = False
leftovers = {'Leftovers added': [], 'Leftovers removed': []}
```

 And with that in mind, it make sense both cases each time, as these are independent and we care about both.